### PR TITLE
feat(data-transfer): data import supports file directory

### PR DESCRIPTION
**Pull request by persistafei**
What type of PR is this?
type-feature

What this PR does / why we need it:
data import now supports choosing file directory

Which issue(s) this PR fixes:
Fixes #

Special notes for your reviewer:
Additional documentation e.g., usage docs, etc.:

### CODE REVIEW

File: server/odc-service/src/main/java/com/oceanbase/odc/service/datatransfer/loader/PlSqlDirOutput.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+import org.apache.commons.io.FileUtils;
+
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * @author youshu
+ * @date 2024/8/27
+ */
+@Slf4j
+public class PlSqlDirOutput extends AbstractThirdPartyOutput {
```
Comment (yizhouxw) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/datatransfer/loader/PlSqlDirOutput.java:30]: PLSqlDirectoryOutput

File: server/odc-service/src/main/java/com/oceanbase/odc/service/datatransfer/loader/PlSqlMultiFileOutput.java [hunk 2]
```diff
@@ -29,27 +27,11 @@
 import org.apache.commons.compress.utils.IOUtils;
 
 import com.oceanbase.odc.common.file.zip.ZipFileTree;
-import com.oceanbase.odc.common.util.StringUtils;
 
 import lombok.extern.slf4j.Slf4j;
 
 @Slf4j
 public class PlSqlMultiFileOutput extends AbstractThirdPartyOutput {
```
Comment (yizhouxw) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/datatransfer/loader/PlSqlMultiFileOutput.java:34]: PLSqlMultiFileOutput

File: server/odc-service/src/main/java/com/oceanbase/odc/service/datatransfer/DataTransferService.java [hunk 3]
```diff
@@ -229,8 +229,8 @@ public DataTransferTaskContext create(@NonNull String bucket,
     public UploadFileResult getMetaInfo(@NonNull String fileName) throws IOException {
         File uploadFile = fileManager.findByName(TaskType.IMPORT, LocalFileManager.UPLOAD_BUCKET, fileName).orElseThrow(
                 () -> new FileNotFoundException("File not found"));
-        if (!uploadFile.exists() || !uploadFile.isFile()) {
-            throw new IllegalArgumentException("Target is not a file or does not exist, " + fileName);
+        if (!uploadFile.exists()) {
```
Comment (yizhouxw) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/datatransfer/DataTransferService.java:232]: we may add a parameter for distinguish import by file or directory

Comment (persistafei) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/datatransfer/DataTransferService.java:238]: added param:  fileType

**Review by yizhouxw — APPROVED** [review]

**Review by LuckyPickleZZ — APPROVED** [review]
lgtm

**Pull request merged**